stdlib/public/core/LazyCollection.swift

```diff
@@ -10,8 +10,32 @@
 //
 //===----------------------------------------------------------------------===//
 
-@available(swift, deprecated: 5.0, renamed: "LazySequence")
-public typealias LazyCollectionProtocol = LazySequenceProtocol
+public protocol _LazyCollectionProtocol: Collection, LazySequenceProtocol {		
+   /// A `Collection` that can contain the same elements as this one,		
+   /// possibly with a simpler type.		
+   ///		
+   /// - See also: `elements`		
+   associatedtype Elements : Collection = Self		
+}
+
+extension _LazyCollectionProtocol {		
+   // Lazy things are already lazy		
+   @inlinable // protocol-only		
+   public var lazy: LazyCollection<Elements> {		
+     return elements.lazy		
+   }		
+ }		
+		
+ extension _LazyCollectionProtocol where Elements: _LazyCollectionProtocol {		
+   // Lazy things are already lazy		
+   @inlinable // protocol-only		
+   public var lazy: Elements {		
+     return elements		
+   }		
+ }
+
+@available(swift, deprecated: 5.0, message: "Prefer LazySequence where Self (or Elements): Collection")
+public typealias LazyCollectionProtocol = _LazyCollectionProtocol
```

Comment:
Why have the underscored name at all?

Reply:
to avoid warnings when the std lib conforms types to it.

Reply:
OK I relent, no underscore then. I've put it back without deprecation for now, since the deprecation can come after ABI.